implementations/rust/ockam/ockam_identity/src/secure_channel/handshake/handshake_worker.rs

```diff
@@ -133,7 +133,18 @@ impl Worker for HandshakeWorker {
 
     async fn shutdown(&mut self, _context: &mut Self::Context) -> Result<()> {
         if let Some(handler) = &self.decryptor_handler {
-            handler.shutdown().await?
+            let encryptor_address = &handler.addresses.encryptor;
+            if let Some(entry) = self.secure_channels.secure_channel_registry.unregister_channel(encryptor_address) {
+                let err = _context.stop_worker(entry.decryptor_messaging_address().clone()).await.err();
+                if let Some(err) = err {
+                    return Err(err);
+                }
+            } else {
+                return Err(IdentityError::SecureChannelNotFound.into());
+            }
+
+            _context.stop_worker(handler.addresses.encryptor.clone()).await?;
```

Comment:
Let's just attempt to stop that Address either `self.decryptor_handler` is Some or None. And let's ignore the error.